src/mount/mount-tool.c

```diff
@@ -372,51 +399,50 @@ static int parse_argv(int argc, char *argv[]) {
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                "At most two arguments required.");
 
-                if (arg_mount_type && !fstype_is_blockdev_backed(arg_mount_type)) {
-                        arg_mount_what = strdup(argv[optind]);
+                if (strv_contains(arg_property, "Type=tmpfs")) {
```

Comment:
i find this a bit too automatic.

What about instead just adding `systemd-mount --tmpfs`? or even `systemd-mount -T` for this?

That would be much closer to your original patch, you'd just add two dashes to your original command line, i.e. your `systemd-mount tmpfs foo` becomes `systemd-mount --tmpfs foo`. And if you use the `-T` short version it even becomes three chars shorter than your original cmdline

Reply:
sounds good to me